lib/y2_installbase.pm

```diff
@@ -264,18 +291,37 @@ sub save_strace_gdb_output {
 
 sub post_fail_hook {
     my $self = shift;
-    $self->SUPER::post_fail_hook;
-    get_to_console;
-    $self->detect_bsc_1063638;
-    $self->get_ip_address;
-    $self->remount_tmp_if_ro;
-    # Avoid collectin logs twice when investigate_yast2_failure() is inteded to hard-fail
-    $self->save_upload_y2logs unless get_var('ASSERT_Y2LOGS');
-    return if is_caasp;
-    $self->save_system_logs;
-
-    # Collect yast2 installer  strace and gbd debug output if is still running
-    $self->save_strace_gdb_output;
+    if (check_var("REMOTE_CONTROLLER", "vnc")) {
+        wait_screen_change { send_key 'ctrl-alt-shift-x' };
+        $self->save_remote_upload_y2logs unless get_var('ASSERT_Y2LOGS');
+    }
+    if (check_var("REMOTE_CONTROLLER", "ssh")) {
+        my $remote_ip = get_var('TARGET_IP');
```

Comment:
Have you cross-checked if we can reuse `root-ssh` console?

Reply:
i think i tried this in some point without success. but it might have not been when the remote_controller was ssh. For sure it didnt work for VNC.

Reply:
for VNC, we might still enable ssh, so that both are available. The thing is that will solve issue with timeouts I've mentioned above. Like on zVMs we have hypervison in between and able to collect logs without having errors: https://openqa.suse.de/tests/2975434#step/logs_from_installation_system/2 It works pretty easy, on the worker itself we start xterm and run commands there, so have access to the return codes which can validate.
Your solution works functionally, but unfortunately solves the problem for single scenario.
So I guess we should take a closer look there, as then we solve it for VNC scenarios too.

Reply:
`root-ssh` doesnt work. http://aquarius.suse.cz/tests/1332#

Reply:
That's clear that it won't work straight away. Question is why and if we can adjust it for this scenario.

Reply:
First thing i can see is
```
[2019-06-17T15:21:04.785 CEST] [debug] <<< testapi::select_console(testapi_console='root-ssh')
console root-ssh does not exist at /usr/lib/os-autoinst/backend/driver.pm line 95.
[2019-06-17T15:21:04.785 CEST] [debug] Backend process died, backend errors are reported below in the following lines:
Can't call method "select" on an undefined value at /usr/lib/os-autoinst/backend/baseclass.pm line 556.
```
but i guess i can dig more